### PR TITLE
help: display external subcommand help

### PR DESCRIPTION
**Pull request by euclio**
When invoking `cargo help <subcommand>`, if the subcommand isn't found, but it *is* an external subcommand, call that subcommand with `--help`.

A test should probably be written for this, but I'm not sure how best to mock an external subcommand.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR! I think you should be able to emulate a custom subcommand similar to `install.rs` how custom subcommands are installed?

**Comment by euclio** [comment]
@alexcrichton Added test.

File: src/bin/cargo/cli.rs [hunk 1]
```diff
@@ -9,7 +9,19 @@ use super::commands;
 use command_prelude::*;
 
 pub fn main(config: &mut Config) -> CliResult {
-    let args = cli().get_matches_safe()?;
+    let args = match cli().get_matches_safe() {
+        Ok(args) => args,
+        Err(e) => {
+            if e.kind == clap::ErrorKind::UnrecognizedSubcommand {
+                // An unrecognized subcommand might be an external subcommand.
+                let cmd = &e.info.as_ref().unwrap()[0].to_owned();
```
Comment (matklad) [inline src/bin/cargo/cli.rs]: Reading clap's docs, it is not clear that this wont panic on unwrap or `[0]` in some cases. But let's leave this as is: this is an error-path anyway, so loud panic should be not cause problems, and would be easier to fix than slient do-nothing.

File: src/bin/cargo/cli.rs [hunk 2]
```diff
@@ -9,7 +9,19 @@ use super::commands;
...
+                return super::execute_external_subcommand(config, cmd, &[cmd, "--help"])
```
Comment (matklad) [inline src/bin/cargo/cli.rs]: This assumes that subcommands support `--help` argument. I think this is a reasonable assumption, but let's add 

> Cargo assumes that `cargo-${command} --help` prints a help message

line to docs here: https://github.com/rust-lang/cargo/blob/master/src/doc/src/reference/external-tools.md#custom-subcommands

Comment (euclio) [reply src/bin/cargo/cli.rs]: Added a small section about this assumption, and cleaned up the docs around it a bit.

**Comment by matklad** [comment]
LGTM!

@alexcrichton anything else to add?

**Pull request merged**